### ISSUE TITLE
REST endpoint for configured and supported WebAPI features

### ISSUE THREAD

**Issue by pavgra** [OHDSI/WebAPI#303]
During last architecture call (11 Jan) there was a discussion, that it would be useful to have REST endpoint, which returns what features are supported by current instance of WebAPI. E.g. whether authentication is turned on / off, what authentication providers are configured and available, whether there are any Data sources with CDM v5.3 connected to the WebAPI and therefore whether it is reasonable to show payer_plan options in cohort builder (the ones which are only supported by v5.3 and at older models will just cause errors), etc. Because now we have to sync such things between back-end and front-end configuration, which is bad approach - no single source of truth.

**Comment by chrisknoll** [comment]
There is a REST endpoint off webapi root called /info which currently just returns the version of WebAPI. This could be extended to include other meta about the features enabled on Webapi such as security.

Separately, when considering what the CDM version is, that's something related to a source, not WebAPI,  Presumably you could have multiple sources registered with webapi and those sources could have different versions of the CDMs.  Were you thinking that when you go to the rest endpionts, it could return the set of CDM versions that are registered as sources?

**Comment by t-abdul-basser** [comment]
@pavgra  This would be very welcome 
@chrisknoll I agree that extending existing /info endpoint rather than adding new endpoint is preferred implementation approach.

**Comment by pavgra** [comment]
@chrisknoll, I totally agree with your statement, that there are usually multiple sources. My idea was that e.g. if there are no sources with CDM 5.3, it makes no sense to show not working filtering critereas in cohort builder

**Comment by chrisknoll** [comment]
Hi, @pavgra , is there plans to update the cohort editor to be CDM-aware? We tried to do something like this in Achilles (where we had a v4 and v5 versions) and in the end it was not maintainable.  Were you thinking along these lines?

**Comment by pavgra** [comment]
@chrisknoll, the thing, which I would like to start from is displaying auth configuration in that endpoint. But then, I guess, it would be good to move into the direction you mentioned - make cohort-editor to be CDM-aware. I think, that each filter could be marked with CDM version, which it requires, so we would be able, to at least, calculate and highlight in UI, what is the minimal CDM version required for the cohort, and then in generation tab to disable "Generate" button, if the source doesn't match minimum supported CDM version.

**Comment by anthonysena** [comment]
Hi - perhaps it is worth splitting these ideas into 2 features: 1 to track the changes to the WebAPI/info to expose information regarding version info, auth info, etc and another issue to think through how we make WebAPI/Atlas aware of differences in CDM version specifications?

**Comment by chrisknoll** [comment]
@pavgra , I've proposed a change tot he circe expression to provide a place for a semver expression to declare the allowed cdm version here: https://github.com/OHDSI/circe-be/issues/27.  If you have any comments on this, please share them under that issue.  This is only a proposal, nothing has been decided yet.

**Comment by fdefalco** [comment]
I think this all makes sense.  We will need to extend the data source configuration we have in the ohdsi schema to include the version of the CDM schema used by the data source.  Or is this a daimon level configuration since CDM is a daimon type? Open to thoughts.
